src/schedule.py

```diff
@@ -86,20 +88,97 @@ def cfg_to_dfg(cfg, graphs, latency):
 
     return cfg_node_to_dfg_map
 
+def assign_time_of_execution(graph):
+    """
+    Calculates when each operation will take place,
+    overwriting the "dist" attribute from before.
+    """
+    for node in graph.nodes:
+        graph.nodes[node]["dist"] =  0
+    graph = nx.reverse(graph)
+    for gen in list(nx.topological_generations(graph))[1:]:
+        max_weight = 0
+        for node in gen:
+            in_edge = list(graph.in_edges(node))[0]
+            max_weight = max(max_weight, graph.edges[in_edge]["weight"])
+        for node in gen:
+            in_edge = list(graph.in_edges(node))[0]
+            graph.edges[in_edge]["weight"] = max_weight
```

Comment:
I'm also failing to understand why all edges in a particular layer have to have the same weight

Reply:
has to do with graph based analysis stuff we discussed with prof. mitra and suhyeong last quarter, i.e. the operations in a layer have to wait for each other before proceeding. Otherwise you run into a case where add operation finishes before parallel register accesses and moves onto a subsequent register access, thereby causing 3 registers to be in use when we only have 2 in the hardware (I had observed this in matmult).